src/packages/client/src/__tests__/integration/errors/wrong-native-types-mysql/test.ts

```diff
@@ -0,0 +1,199 @@
+import { getTestClient } from '../../../../utils/getTestClient'
+import path from 'path'
+import { migrateDb } from '../../__helpers__/migrateDb'
+import Decimal from 'decimal.js'
+import { tearDownMysql } from '../../../../utils/setupMysql'
+
+beforeAll(async () => {
+  process.env.TEST_MYSQL_URI += '-wrong-native-types'
+  await tearDownMysql(process.env.TEST_MYSQL_URI!)
+  await migrateDb({
+    connectionString: process.env.TEST_MYSQL_URI!,
+    schemaPath: path.join(__dirname, 'schema.prisma'),
+  })
+})
+
+test('wrong-native-types-mysql A: Int, SmallInt, TinyInt, MediumInt, BigInt', async () => {
+  const PrismaClient = await getTestClient()
+
+  const prisma = new PrismaClient({ errorFormat: 'minimal' })
+
+  await prisma.a.deleteMany()
+
+  const data = {
+    int: 123,
+    sInt: 12,
+    tInt: 1,
+    mInt: 100,
+    bInt: 123123123.1,
+  }
+
+  await expect(async () =>
+    prisma.a.create({
+      data,
+      select: {
+        int: true,
+        sInt: true,
+        mInt: true,
+        bInt: true,
+        tInt: true,
+      },
+    }),
+  ).rejects.toThrowErrorMatchingInlineSnapshot(`
+          Argument tInt: Got invalid value 1 on prisma.createOneA. Provided Int, expected Boolean.
+          Argument bInt: Got invalid value 123123123.1 on prisma.createOneA. Provided Float, expected Int.
+
+        `)
+
+  prisma.$disconnect()
+})
+
+test('wrong-native-types-mysql B: Float, Double, Decimal, Numeric', async () => {
+  const PrismaClient = await getTestClient()
+
+  const prisma = new PrismaClient({ errorFormat: 'minimal' })
+
+  await prisma.b.deleteMany()
+
+  let data: any = {
+    float: 12.2,
+    dFloat: 10.2,
+    decFloat: 1.1,
+    numFloat: 'a5.6',
+  }
+
+  await expect(async () =>
+    prisma.b.create({
+      data,
+      select: {
+        float: true,
+        dFloat: true,
+        decFloat: true,
+        numFloat: true,
+      },
+    }),
+  ).rejects.toThrowErrorMatchingInlineSnapshot(`
+          Argument numFloat: Got invalid value 'a5.6' on prisma.createOneB. Provided String, expected Decimal or Null.
+
+        `)
+
+  prisma.$disconnect()
+})
+
+test('wrong-native-types-mysql C: Char, VarChar, TinyText, Text, MediumText, LongText', async () => {
+  const PrismaClient = await getTestClient()
+
+  const prisma = new PrismaClient({ errorFormat: 'minimal' })
+
+  await prisma.c.deleteMany()
+
+  const data = {
+    char: 'f0f0f0f0f20',
+    vChar: '123456789012',
+    tText: 'f'.repeat(258),
+    text: 'l'.repeat(70_000),
+    mText: '🥳'.repeat(70_000),
+    lText: '🔥'.repeat(80_000),
+  }
+
+  await expect(async () =>
+    prisma.c.create({
+      data,
+      select: {
+        char: true,
+        vChar: true,
+        tText: true,
+        mText: true,
+        text: true,
+        lText: true,
+      },
+    }),
+  ).rejects.toThrowErrorMatchingInlineSnapshot(
+    `The provided value for the column is too long for the column's type. Column: char`,
```

Comment:
Nice!

Reply:
So we only get an error for `char`, I think `tText` should error too but probably needs to be done separately?

Reply:
Yeah, for now I think it's ok to have one error case to catch them all, we can later invest in more